gestorpsi/client/tests.py

```diff
@@ -20,33 +20,52 @@
 
 from gestorpsi.client.models import Client, ClientManager
 from gestorpsi.client.models import Person
+from gestorpsi.service.models import Service, ServiceType, Area
 from gestorpsi.phone.models import Phone
-from gestorpsi.util.test_utils import user_stub, setup_required_data
+from gestorpsi.admission.models import ReferralChoice as AdmissionChoice
+from gestorpsi.organization.models import Organization
+from gestorpsi.authentication.views import login_check
+from gestorpsi.util.test_utils import client_stub, user_stub, setup_required_data, setup_required_service
 
 class ClientTests(TestCase):
     def setUp(self):
         self.factory = RequestFactory()
         # OBS: Using cl instead of client to evade conflicts with Client class
-        self.cl = Cl()
+        self.client = Cl()
         setup_required_data()
+        self.name_client = 'Cliente 10'
 
     def test_client_index_should_redirect_unlogged_user(self):
-        req = self.cl.get(reverse('client-index'))
+        req = self.client.get(reverse('client-index'))
         self.assertEqual(req.status_code, 302)
 
     def test_client_index_should_work_for_logged_user(self):
         # user creation and login
-        self.cl.post(reverse('registration-register'), user_stub())
-        self.cl.login(
-            username=user_stub()["username"],
-            password=user_stub()["password1"]
-        )
+        self.client.post(reverse('registration-register'), user_stub())
+        res = self.client.login(username=user_stub()["username"], password=user_stub()["password1"])
         user = User.objects.get(username=user_stub()["username"])
         user.is_superuser = True
         user.save()
-        req = self.cl.get(reverse('client-index'))
+        req = self.client.get(reverse('client-index'))
         self.assertEqual(req.status_code, 200)
 
+    def test_client_add(self):
+        self.client.post(reverse('registration-register'), user_stub())
+        res = self.client.login(username=user_stub()["username"], password=user_stub()["password1"])
+        user = User.objects.get(username=user_stub()["username"])
+        user.is_superuser = True
+        user.save()
+
+        setup_required_service()
+
+        response = self.client.post(reverse('client-form-new'))
+        response = self.client.post(reverse('client-form-save'), client_stub())
+
+        client = Client.objects.all()[0]
```

Comment:
using `client` as the variable that refers to the gestorpsi client model and `client` to the request manager can be error prone.